Services/Simulations.cs

```diff
@@ -367,6 +379,82 @@ public async Task DeleteAsync(string id)
             await this.simulationsStorage.DeleteAsync(id);
         }
 
+        public async Task TrySeedAsync(string templateName)
+        {
+            try
+            {
+                // Skips if the seeding has already been complete
+                if (!await this.mainStorage.ExistsAsync(SEED_STATUS_KEY))
+                {
+                    await this.SeedSimulationsAsync(templateName);
+                    var record = new StorageRecord { Id = SEED_STATUS_KEY, Data = "Seed Completed" };
```

Comment:
Line 390, 391 should be moved inside SeedSimulationsAsync. We should only set the key to "Seed Completed" if the creation successfully completed. In case of an exception (catch block in SeedSimulationsAsync) this should be set to some error state.

Reply:
We don't need to set this key in case of error. We should set it only if this completes successfully.

Reply:
But this will set the key even when SeedSimulationsAsync fails - the exception will be caught in catch block in SeedSimulationsAsync.

Reply:
Throw from catch block in SeedSimulationsAsync should handle this case.